src/rag/embedding.py

```
@@ -0,0 +1,88 @@
+        try:
+            result = self.client.models.embed_content(
+                model=self.model_name,
+                contents=batch, # type: ignore
+                config=types.EmbedContentConfig(
+                    task_type=task_type.value,
+                    output_dimensionality=self.dimension,
+                ),
+            )
+                
+            if not result.embeddings:
+                print(f"Batch returned no embeddings. Filling with zeros.")
```

Ruff (0.14.14)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)